src/agent/solana/exporter.rs

```diff
@@ -321,6 +334,21 @@ impl Exporter {
     }
 
     async fn publish_batch(&self, batch: &[(&Identifier, &PriceInfo)]) -> Result<()> {
+        let publish_keypair = if let Some(kp) = self.key_store.publish_keypair.as_ref() {
+            // It's impossible to sanely return a &Keypair in the
+            // other if branch, so we clone the reference.
+            Keypair::from_bytes(&kp.to_bytes())
+                .context("INTERNAL: Could not convert keypair to bytes and back")?
+        } else {
+            debug!(
+                self.logger,
+                "Exporter: Publish keypair is None, requesting remote loaded key"
+            );
+            let kp = RemoteKeypairLoader::request_keypair(&self.keypair_request_tx).await?;
+            debug!(self.logger, "Exporter: Keypair received");
+            kp
+        };
+
```

Comment:
I think it is better to drop the request if it cannot get the key because when in the current code when it gets the key the api will bloat the rpc.